frontend/catalyst/passes.py

```diff
@@ -137,15 +139,29 @@ def g(x: float):
         raise TypeError(f"A QNode is expected, got the classical function {fn}")
 
     wrapped_qnode_function = fn.func
+    funcname = fn.__name__
 
     def wrapper(*args, **kwrags):
+        # TODO: hint the compiler which qnodes to run the pass on via an func attribute,
+        # instead of the qnode name. That way the clone can have this attribute and
+        # the original can just not have it.
+        # We are not doing this right now and passing by name because this would
+        # be a discardable attribute (i.e. a user/developer wouldn't know that this
+        # attribute exists just by looking at qnode's documentation)
+        # But when we add the full peephole pipeline in the future, the attribute
+        # could get properly documented.
+
         apply_registered_pass_p.bind(
-            pass_name="remove-chained-self-inverse", options=f"func-name={fn.__name__}"
+            pass_name="remove-chained-self-inverse",
+            options=f"func-name={funcname}" + "_cancel_inverses",
```

Comment:
I think this will blow up the symbol pretty quickly with longer pipelines but for now it's probably fine 😅

Reply:
Yeah, this is why the comment above suggests using some sort of attribute. We can discuss more when we start working on the entire pipeline frontend